### PR TITLE
feat(menus): update Menu/Select Item indentation and focus based on item hover

### PR DESCRIPTION
**Pull request by austin94**
* [x] **BREAKING CHANGE** - `HeaderItem` styling changes and new focus/hover logic

Should also force a breaking change onto all dependent packages; rather than the default lerna behavior of marking it as a patch.

## Description

- Removes an invalid `active` prop from a Menu example
- Adds `containsIcon` property for `HeaderItem` components (`icon` is a native attribute 😢 )
- Adds `onMouseMove` event to allow "focus on hover" mechanism
  - @allisonacs lets meet-up and go over this interaction pattern to double-check a few things

## Checklist

* [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
* [x] :nail_care: view component styling is based on a Garden CSS
  component
* [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :guardsman: includes new unit and snapshot tests
* [x] :ledger: any new files are included in the packages `src/index.js` export
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/menus/src/containers/MenuContainer.js [hunk 1]
```diff
@@ -434,6 +434,16 @@ class MenuContainer extends ControlledComponent {
     return {
       key,
       role,
+      /**
+       * onMouseMove is used as it is only triggered by actual mouse movement
+       */
+      onMouseMove: composeEventHandlers(onMouseMove, () => {
```
[the remark this answers is not in the document]
Comment (austin94) [reply packages/menus/src/containers/MenuContainer.js]: I originally had `onMouseMove` then, checked downshift for inspiration...

```
// onMouseMove is used over onMouseEnter here. onMouseMove
// is only triggered on actual mouse movement while onMouseEnter
// can fire on DOM changes, interrupting keyboard navigation
```

**Comment by austin94** [comment]
I agree with @allisonacs, the dual styling is a little jarring.

This is a side-effect of the `:hover` styling being provided by the css-components.  @jzempel is this possible to remove?

**Comment by jzempel** [comment]
Ah yeah, the CSS styling still needs some work 🙁

**Comment by austin94** [comment]
I've updated the `Item` component with some custom styling and pre-published to https://garden.zendesk.com/react-components/menus/

There is some specificity weirdness, but it seems to be working as expected now. Let me know what you think.

**Comment by allisonacs** [comment]
Is it intentional that the old focus state is being used now?

File: packages/menus/src/views/items/Item.js [hunk 2]
```diff
@@ -29,7 +29,16 @@ const Item = styled.li.attrs({
       [MenuStyles['is-checked']]: props.checked
     })
 })`
+  /* stylelint-disable */
+  ${props =>
+    !props.focused &&
+    `&&&:hover {
```
Comment (jzempel) [inline packages/menus/src/views/items/Item.js]: gnarly, but effective!

**Comment by jzempel** [comment]
@austin94 on the demo page if you expand a menu, scroll it off the page, and click a button to expand a different menu ... the page jumps back to the original menu, collapses it, and the 2nd menu does not expand.

**Comment by austin94** [comment]
> on the demo page if you expand a menu, scroll it off the page, and click a button to expand a different menu

This jumping is due to us returning focus to the element which triggered the menu.  It aligns with most of menu implementations, including Material.



We could tweak this some though. A few options could be:
- Once a menu has been scrolled out of view entirely, close it without returning focus to it's trigger
  - Some complications with this is it depends on how popper.js is configured with it's bounding viewport

**Comment by austin94** [comment]
> Is it intentional that the old focus state is being used now?

No, since I was doing a manual publish a fresh `yarn install` didn't happen. This is now updated at https://garden.zendesk.com/react-components/menus/

**Comment by jzempel** [comment]
@austin94 consider a UI like dropbox (web). Expand one of the file menus in the list and then scroll away and expand a different file menu. You _definitely_ wouldn't want the page to scroll back to the originally expanded menu – that would be a bad user experience. Similarly, we wouldn't want to see this kind of experience in a Zendesk table.

**Pull request merged**